### ISSUE TITLE
TypevarTuple is incompatible with classmethod

### ISSUE THREAD

**Issue by RobertCraigie** [microsoft/pyright#3083]
**Describe the bug**

```py
from typing import Tuple
from typing_extensions import Unpack, TypeVarTuple

Shape = TypeVarTuple('Shape')

class Foo:
    @classmethod
    def foo(cls, *shape: Unpack[Shape]) -> Tuple[Unpack[Shape]]:
        ...
```

```
5:6 - error: Argument of type "(cls: Type[Self@Foo], *shape: *Shape@foo) -> Tuple[*Shape@foo]" cannot be assigned to parameter "__f" of type "(...) -> _R_co@classmethod" in function "__init__"
    Type "(cls: Type[Self@Foo], *shape: *Shape@foo) -> Tuple[*Shape@foo]" cannot be assigned to type "(...) -> _R_co@classmethod"
      Parameter 1: type "*tuple[Any, ...]" cannot be assigned to type "*tuple[Type[Self@Foo], *Shape@foo]"
        TypeVarTuple cannot be bound to a tuple of unknown length (reportGeneralTypeIssues)
```

**Expected behavior**

There should be no errors reported.

**Additional context**

Pyright correctly evaluates the return type when used:

```py
f = Foo.foo('a', 1)
reveal_type(f)  # T: Tuple[str, int]
```

**VS Code extension or command-line**

Command-line: `1.1.222`

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.223, which I just published. It will also be included in the next release of pylance.

**Comment by sam-goodwin** [comment]
I'm still running into this on `pyright==1.1.358`

![image](https://github.com/microsoft/pyright/assets/38672686/a124d200-8966-484e-9bf9-4954ba3adea6)

**Comment by erictraut** [comment]
@sam-goodwin, are you able to repro the problem that was reported at the top of this thread? That is, if you copy and paste that code sample exactly, do you see the reported error? If not, then the problem you're seeing is different, and you should open a new issue with repro steps.

**Comment by sam-goodwin** [comment]
But, my error is unrelated to `TypeVarTuple`, so want me to open a new issue?

**Comment by sam-goodwin** [comment]
https://github.com/microsoft/pyright/issues/7662
